mtg_card_identifier/CardScanner/CardScannerFile.swift

```diff
@@ -0,0 +1,159 @@
+        let partialTransparentView = PartialTransparentView(cutout: style.overlay)
+        partialTransparentView.translatesAutoresizingMaskIntoConstraints = true
+        partialTransparentView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        partialTransparentView.frame = vc.overlayContainerView.bounds
+        
+        vc.view.addSubview(partialTransparentView)
+        vc.regionOfInterest = style.regionOfInterest
+        vc.delegate = context.coordinator
+        
+        do {
+            try vc.startScanning()
+            
+        } catch {
+            
+            print("Error: \(error.localizedDescription)")
+        }
+        
+        
+        return vc
+    }
+    
+    func updateUIViewController(_ uiViewController: DataScannerViewController, context: Context) {
+    }
+    
+    @MainActor
+    class Coordinator: NSObject, DataScannerViewControllerDelegate {
+        var scanedTextBinding: Binding<[String]>
+        
+        init(scanedTextBinding: Binding<[String]>) {
+            self.scanedTextBinding = scanedTextBinding
+        }
+        
+        
+        func dataScanner(_ dataScanner: DataScannerViewController, didAdd addedItems: [RecognizedItem], allItems: [RecognizedItem]) {
+            handleItems(allItems)
+        }
+        
+        func dataScanner(_ dataScanner: DataScannerViewController, didUpdate updatedItems: [RecognizedItem], allItems: [RecognizedItem]) {
+            handleItems(allItems)
+        }
+        
+        func handleItems(_ items: [RecognizedItem]) {
+            print("Items count : \(items.count)")
+            scanedTextBinding.wrappedValue = items.compactMap { $0.text }
+            items.forEach { item in
+                switch item {
+                case .text(let text):
+                    print("Scanned words: \(text.transcript)")
+                default: break
+                }
+            }
+        }
+    }
+}
+
+
+class PartialTransparentView: UIView {
+    var cutout: CGRect?
+    
+    init(cutout: CGRect?) {
+        super.init(frame: .zero)
+        self.cutout = cutout
+        backgroundColor = UIColor.black.withAlphaComponent(0.6)
+        isOpaque = false
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func draw(_ rect: CGRect) {
+        backgroundColor?.setFill()
+        UIRectFill(rect)
+        guard let cutout = cutout else { return }
+        
+        
+        let path = UIBezierPath(roundedRect: cutout, cornerRadius: 10)
+        let intersection = rect.intersection(cutout)
+        UIRectFill(intersection)
+        
+        UIColor.clear.setFill()
+        UIGraphicsGetCurrentContext()?.setBlendMode(.copy)
+        path.fill()
+    }
+}
+
+// MARK: Extension
+//            find the window and make the region of interest, init as well, and then return the style of it.
+struct Style {
+    let overlay: CGRect
+    let regionOfInterest: CGRect
```

Comment:
Lets make overlay and regionOfInterest optional
Also, lets make overlay a UIView rather than a CGRect.